src/utils/csv.ts

```diff
@@ -0,0 +1,57 @@
+import { logDebug } from 'utils/miscellaneous'
+
+// TODO: can I make sure - using types - that headers length matches transformer return type length?
+// Alternatively, I could make headers be the keys of a second type parameter
+// which is the result of transformer, such as transformer<T, R>(item: T) => R[]
```

Comment:
Don't see the use for it. Headers can be completely arbitrary

Elaborate how you would do it

Reply:
The idea is that the transformer returns an object instead of a list.
Better for readability and now that I think of it, we might as well get rid of the headers column.

I just don't know about the order in the object.

Let's say I want to convert to csv the following dumb headers, data and transformer:
- headers: ['times1', 'times2', 'times3']
- data: [1, 2, 3]

```ts
transformer<T>(item:T):string[] {
  return [
    item * 1,
    item * 2,
    item * 3
  ]
}
```

Resulting csv should be:
```
times1,times2,times3
1,2,3
2,4,6
3,6,9
```

That's how it's right now. With the amount of columns we have, it's easy to get confused and pass the wrong order/incorrect #of values.

What I'm thinking about is like this, for same data, but without headers and with transformer:

```ts
transformer<T, R>(item:T):R {
  return {
    times1: item*1,
    times2: item*2,
    times3: item*3
}
```

We can extract the headers from the transformer return type `R`.

Now, is the key order guaranteed?

I'm out of time today, might try it out on Monday.

Reply:
The requirement for `R` is that keys and values are strings. There's a way to express that in TS for sure but not right now.

Reply:
:feelsgood: c9b2e38e3ce69343c43a31cba8ac4f5091cec592